### PR TITLE
ath79: some mtd-cal-data removals

### PR DESCRIPTION
**Pull request by neheb**
no idea if this is even right. Completely untested.

### CODE REVIEW

File: target/linux/ath79/dts/qca9531_8dev_lima.dts [hunk 1]
```diff
@@ -80,6 +80,10 @@
 					macaddr_art_6: macaddr@6 {
 						reg = <0x6 0x6>;
 					};
+
+					calibration_art_1000: calibration@1000 {
+						reg = <0x1000 0x800>;
```
Comment (DragonBluep) [inline target/linux/ath79/dts/qca9531_8dev_lima.dts:85]: All SoC integrated ath9k wireless cal data sizes are `0x440`. Include AR933x, AR934x and QCA95xx series.

Comment (DragonBluep) [reply target/linux/ath79/dts/qca9531_8dev_lima.dts:85]: Larger EEPROMs can also be accepted by the ath9k driver. I think the script is indeed incorrect. It doesn't make sense to use different EEPROM sizes for the same chip and the same driver.

Comment (neheb) [reply target/linux/ath79/dts/qca9531_8dev_lima.dts:85]: @andreybondar59 can you verify this?

Comment (neheb) [reply target/linux/ath79/dts/qca9531_8dev_lima.dts:85]: @pepe2k might be able to as well.

Comment (neheb) [reply target/linux/ath79/dts/qca9531_8dev_lima.dts:85]: looking at this again, I think you're right. 

I'll rework this PR and push later.

File: target/linux/ath79/dts/ar7161_meraki_mr16.dts [hunk 2]
```diff
@@ -84,8 +83,8 @@
 		compatible = "pci168c,0029";
 		reg = <0x9000 0 0 0 0>;
 		qca,no-eeprom;
```
Comment (raenye) [inline target/linux/ath79/dts/ar7161_meraki_mr16.dts:85]: Why delete once but not twice?

File: target/linux/ath79/dts/ar9132_tplink_tl-wa901nd-v2.dts [hunk 3]
```diff
@@ -91,10 +91,20 @@
 				reg = <0x020000 0x3D0000>;
 			};
 
-			art: partition@2 {
+			partition@2 {
```
Comment (raenye) [inline target/linux/ath79/dts/ar9132_tplink_tl-wa901nd-v2.dts:94]:
```suggestion
			partition@3f0000 {
```

File: target/linux/ath79/dts/ar9132_tplink_tl-wr1043nd-v1.dts [hunk 4]
```diff
@@ -117,10 +117,20 @@
 				reg = <0x020000 0x7D0000>;
 			};
 
-			art: partition@7F0000 {
+			partition@7F0000 {
```
Comment (raenye) [inline target/linux/ath79/dts/ar9132_tplink_tl-wr1043nd-v1.dts:120]:
```suggestion
			partition@7f0000 {
```
Lowercase?

File: target/linux/ath79/dts/ar9330_dlink_dir-505.dts [hunk 5]
```diff
@@ -101,16 +104,40 @@
...
 				reg = <0x20000 0x10000>;
 				read-only;
+
+				nvmem-layout {
+					compatible = "fixed-layout";
+					#address-cells = <1>;
+					#size-cells = <1>;
+
+					macaddr_mac_4: macaddr@4 {
+						reg = <0x4 0x11>;
```
Comment (raenye) [inline target/linux/ath79/dts/ar9330_dlink_dir-505.dts:138]: Does this work without `mac-base`?

File: target/linux/ath79/dts/ar9330_glinet_gl-ar150.dts [hunk 6]
```diff
@@ -101,7 +101,7 @@
 				reg = <0x050000 0xfa0000>;
 			};
 
-			art: partition@3 {
+			partition@3 {
```
Comment (raenye) [inline target/linux/ath79/dts/ar9330_glinet_gl-ar150.dts:104]:
```suggestion
			partition@ff0000 {
```

File: target/linux/ath79/dts/ar9331_embeddedwireless_dorin.dts [hunk 7]
```diff
@@ -81,7 +81,7 @@
 				reg = <0x050000 0xfa0000>;
 			};
 
-			art: partition@3 {
+			partition@3 {
```
Comment (raenye) [inline target/linux/ath79/dts/ar9331_embeddedwireless_dorin.dts:84]:
```suggestion
			partition@ff0000 {
```

File: target/linux/ath79/dts/ar9331_etactica_eg200.dts [hunk 8]
```diff
@@ -106,7 +106,7 @@
 				reg = <0x50000 0xfa0000>;
 			};
 
-			art: art@ff0000 {
+			art@ff0000 {
```
Comment (raenye) [inline target/linux/ath79/dts/ar9331_etactica_eg200.dts:109]: Should all partitions be called `partition@x`?

File: target/linux/ath79/dts/ar9344_qxwlan_e750x.dtsi [hunk 9]
```diff
@@ -99,10 +99,20 @@
 				};
 			};
 
-			art: partition@60000 {
+			partition@60000 {
 				label = "art";
 				reg = <0x060000 0x010000>;
```
Comment (raenye) [inline target/linux/ath79/dts/ar9344_qxwlan_e750x.dtsi:104]: Maybe drop leading zeros?

File: target/linux/ath79/dts/ar7161_meraki_mr16.dts (hunk 2 above)
Comment (neheb) [reply target/linux/ath79/dts/ar7161_meraki_mr16.dts:85]: whoops

File: target/linux/ath79/dts/ar9331_etactica_eg200.dts (hunk 8 above)
Comment (neheb) [reply target/linux/ath79/dts/ar9331_etactica_eg200.dts:109]: I really don't want to make too many unrelated changes. with this for example, notice how label is missing.

File: target/linux/ath79/dts/ar9344_qxwlan_e750x.dtsi (hunk 9 above)
Comment (neheb) [reply target/linux/ath79/dts/ar9344_qxwlan_e750x.dtsi]: Done.

File: target/linux/ath79/dts/ar9331_embeddedwireless_dorin.dts (hunk 7 above)
Comment (neheb) [reply target/linux/ath79/dts/ar9331_embeddedwireless_dorin.dts]: Done.

File: target/linux/ath79/dts/ar9132_tplink_tl-wr1043nd-v1.dts (hunk 4 above)
Comment (neheb) [reply target/linux/ath79/dts/ar9132_tplink_tl-wr1043nd-v1.dts]: Done.

File: target/linux/ath79/dts/ar9132_tplink_tl-wa901nd-v2.dts (hunk 3 above)
Comment (neheb) [reply target/linux/ath79/dts/ar9132_tplink_tl-wa901nd-v2.dts]: Done.

File: target/linux/ath79/dts/ar7161_meraki_mr16.dts (hunk 2 above)
Comment (neheb) [reply target/linux/ath79/dts/ar7161_meraki_mr16.dts]: Fixed.

File: target/linux/ath79/dts/ar9330_dlink_dir-505.dts (hunk 5 above)
Comment (neheb) [reply target/linux/ath79/dts/ar9330_dlink_dir-505.dts:138]: No idea. ping @rmilecki

Comment (neheb) [reply target/linux/ath79/dts/ar9330_dlink_dir-505.dts:139]: Looks like you're right. Updated.

**Comment by neheb** [comment]
Finally rebased.

@DragonBluep can you check if the first commit is correct?

File: target/linux/ath79/dts/ar9344_pcs_cap324.dts [hunk 10]
```diff
@@ -150,10 +153,10 @@
 &wmac {
 	status = "okay";
 
+	nvmem-cells = <&macaddr_art_0 (-1)>, <&cal_art_5000>;
```
Comment (DragonBluep) [inline target/linux/ath79/dts/ar9344_pcs_cap324.dts:156]: I think the dts file is correct. It seems that wmac always load eeprom from low address `0x1000`. And the PCIe NIC ([AR9382](https://openwrt.org/toh/hwdata/powercloud_systems/powercloud_systems_cap324)) eeprom size should be the same 0x440.

Here is an eeprom collection.
[art.zip](https://github.com/openwrt/openwrt/files/14318197/art.zip)

Comment (neheb) [reply target/linux/ath79/dts/ar9344_pcs_cap324.dts:156]: The implication is that currently, both the wmac and pcie devices use the same calibration.

Comment (DragonBluep) [reply target/linux/ath79/dts/ar9344_pcs_cap324.dts:156]: I found an older patch can be referenced.
https://patchwork-proxy.ozlabs.org/project/openwrt/patch/1450392667-22656-2-git-send-email-openwrt@daniel.thecshore.com/

Comment (neheb) [reply target/linux/ath79/dts/ar9344_pcs_cap324.dts:156]: ping @danielfdickinson

**Comment by danielfdickinson** [comment]
@neheb It has been a long time, but I remember that the radios on CAP324 were reversed from what one would expect. That is the 5Ghz used what was normally the 2.4Ghz calibration and vice versa.

I'm not sure how much I can help now; I'd have to check if I even still have any of these devices to test and I haven't built or used OpenWrt in a few years.

**Comment by neheb** [comment]
Sounds like https://github.com/openwrt/openwrt/commit/a14170b6e9c8bab493c5e0e7689405a8a0675e is wrong then.

File: target/linux/ath79/dts/ar9344_pcs_cap324.dts (hunk 10 above)
Comment (neheb) [reply target/linux/ath79/dts/ar9344_pcs_cap324.dts]: Removed mtd-cal-data from pcie instead. It's not correct anyway.

**Comment by neheb** [comment]
@DragonBluep btrw since you did work on this, know anything about https://github.com/openwrt/openwrt/blob/main/target/linux/ath79/dts/ar9344_dlink_dir-8x5.dtsi#L61 ?

I originally had qca,no-eeprom removals but aborted when I found this. Probably better to keep this PR to just mtd-cal-data removals.

**Comment by DragonBluep** [comment]
> @DragonBluep btrw since you did work on this, know anything about https://github.com/openwrt/openwrt/blob/main/target/linux/ath79/dts/ar9344_dlink_dir-8x5.dtsi#L61 ?
> 
> I originally had qca,no-eeprom removals but aborted when I found this. Probably better to keep this PR to just mtd-cal-data removals.

The previous `mac-address-ascii` implementation doesn't work on some platform driver, see https://github.com/openwrt/openwrt/pull/9780#issuecomment-1486188837. It should be fine now since we have ported it to `mac-base` nvmem layout. At least I can sure mac-address nvmem cell works on ar9287 and ar938x ath9k PCIe devices.

**Comment by neheb** [comment]
OK, made https://github.com/openwrt/openwrt/pull/14666 for that.

**Comment by neheb** [comment]
Rebased.

**Comment by neheb** [comment]
ping @Ansuel 

This is ready. Problematic changes were split into a separate commit.

File: target/linux/ath79/dts/ar1022_sitecom_wlr-7100.dts [hunk 11]
```diff
@@ -147,7 +147,7 @@
 				read-only;
 			};
 
-			art: partition@7f0000 {
+			partition@7f0000 {
```
Comment (DragonBluep) [inline target/linux/ath79/dts/ar1022_sitecom_wlr-7100.dts:150]: Hi! Please move these changes to a different commit.

File: target/linux/ath79/dts/ar9132_tplink_tl-wa901nd-v2.dts [hunk 12]
```diff
@@ -87,16 +87,26 @@
 				};
 			};
 
-			partition@1 {
+			partition@20000 {
```
Comment (DragonBluep) [inline target/linux/ath79/dts/ar9132_tplink_tl-wa901nd-v2.dts:90]: This should also be moved to a new commit.

**Review by DragonBluep — COMMENTED** [review]
Would you mind rebase it. I remember recent commits have introduced some new `mtd-cal-data` properties.

File: target/linux/ath79/dts/ar1022_sitecom_wlr-7100.dts (hunk 11 above)
Comment (neheb) [reply target/linux/ath79/dts/ar1022_sitecom_wlr-7100.dts:150]: hmm? Do you mean the standalone art: removals?

Comment (DragonBluep) [reply target/linux/ath79/dts/ar1022_sitecom_wlr-7100.dts:150]: Yes. This file does not contain nvmem changes, so it should be a separate commit.

File: target/linux/ath79/dts/qca9563_qxwlan_e1700ac.dtsi [hunk 13]
```diff
@@ -113,6 +113,10 @@
 					#address-cells = <1>;
 					#size-cells = <1>;
 
+					cal_art_1000: calibration@1000 {
```
Comment (DragonBluep) [inline target/linux/ath79/dts/qca9563_qxwlan_e1700ac.dtsi:116]: Move it to nvmem commit.

File: target/linux/ath79/dts/ar7161_buffalo_wzr-hp-ag300h.dtsi [hunk 14]
```diff
@@ -146,7 +146,7 @@
 				read-only;
 			};
 
-			art: partition@50000 {
+			partition@50000 {
```
Comment (DragonBluep) [inline target/linux/ath79/dts/ar7161_buffalo_wzr-hp-ag300h.dtsi:149]: Move it to `label` commit.

File: target/linux/ath79/dts/ar7240_netgear_wnr612-v2.dtsi [hunk 15]
```diff
@@ -101,7 +101,7 @@
 				label = "firmware";
 			};
 
-			art: partition@3f0000 {
+			partition@3f0000 {
```
Comment (DragonBluep) [inline target/linux/ath79/dts/ar7240_netgear_wnr612-v2.dtsi:104]: Move it to `label` commit.

File: target/linux/ath79/dts/ar7240_tplink.dtsi [hunk 16]
```diff
@@ -99,7 +99,7 @@
 				label = "firmware";
 			};
 
-			art: partition@3f0000 {
+			partition@3f0000 {
```
Comment (DragonBluep) [inline target/linux/ath79/dts/ar7240_tplink.dtsi:102]: Move it to `label` commit.

File: target/linux/ath79/dts/ar7242_ubnt_sw.dtsi [hunk 17]
```diff
@@ -100,7 +100,7 @@
 				read-only;
 			};
 
-			art: partition@7f0000 {
+			partition@7f0000 {
```
Comment (DragonBluep) [inline target/linux/ath79/dts/ar7242_ubnt_sw.dtsi:103]: Move it to `label` commit.

File: target/linux/ath79/dts/ar724x_ubnt_xm.dtsi [hunk 18]
```diff
@@ -58,7 +58,7 @@
 				read-only;
 			};
 
-			art: partition@7f0000 {
+			partition@7f0000 {
```
Comment (DragonBluep) [inline target/linux/ath79/dts/ar724x_ubnt_xm.dtsi:61]: Move it to `label` commit.

File: target/linux/ath79/dts/ar9341_engenius_eap300-v2.dts [hunk 19]
```diff
@@ -56,7 +56,8 @@
 &wmac {
 	status = "okay";
 
-	mtd-cal-data = <&art 0x1000>;
+	nvmem-cells = <&cal_art_1000>;
+	nvmem-cell-names = "mac-address", "calibration";
```
Comment (DragonBluep) [inline target/linux/ath79/dts/ar9341_engenius_eap300-v2.dts:60]: nit.

File: target/linux/ath79/dts/ar9342_ubnt_xw.dtsi [hunk 20]
```diff
@@ -62,7 +62,7 @@
 				read-only;
 			};
 
-			art: partition@7f0000 {
+			partition@7f0000 {
```
Comment (DragonBluep) [inline target/linux/ath79/dts/ar9342_ubnt_xw.dtsi:65]: Move it to `label` commit.

File: target/linux/ath79/dts/ar9342_zyxel_nwa11xx.dtsi [hunk 21]
```diff
@@ -111,7 +111,7 @@
 				};
 			};
 
-			art: partition@ff0000 {
+			partition@ff0000 {
```
Comment (DragonBluep) [inline target/linux/ath79/dts/ar9342_zyxel_nwa11xx.dtsi:114]: Move it to `label` commit.

File: target/linux/ath79/dts/ar9344_compex_wpj344-16m.dts [hunk 22]
```diff
@@ -98,10 +98,28 @@
...
+					#address-cells = <1>;
+					#size-cells = <1>;
+
+					macaddr_art_0: macaddr@0 {
+						reg = <0x0 0x6>;
+					};
+
+					macaddr_art_6: macaddr@6 {
+						reg = <0x6 0x6>;
+					};
```
Comment (DragonBluep) [inline target/linux/ath79/dts/ar9344_compex_wpj344-16m.dts:117]: Useless mac address cells?

File: target/linux/ath79/dts/ar9344_dlink_dir-8x5.dtsi [hunk 23]
```diff
@@ -135,7 +135,7 @@
 				};
 			};
 
-			art: partition@ff0000 {
+			partition@ff0000 {
```
Comment (DragonBluep) [inline target/linux/ath79/dts/ar9344_dlink_dir-8x5.dtsi:138]: Move it to `label` commit.

File: target/linux/ath79/dts/ar9344_qxwlan_e750x.dtsi [hunk 24]
```diff
@@ -101,10 +101,20 @@
 				};
 			};
 
-			art: partition@60000 {
+			partition@60000 {
 				label = "art";
-				reg = <0x060000 0x010000>;
+				reg = <0x60000 0x010000>;
```
Comment (DragonBluep) [inline target/linux/ath79/dts/ar9344_qxwlan_e750x.dtsi:106]: unnecessary change?

File: target/linux/ath79/dts/ar9344_tplink_tl-wdrxxxx.dtsi [hunk 25]
```diff
@@ -103,7 +103,7 @@
 				reg = <0x020000 0x7d0000>;
 			};
 
-			art: partition@7f0000 {
+			partition@7f0000 {
```
Comment (DragonBluep) [inline target/linux/ath79/dts/ar9344_tplink_tl-wdrxxxx.dtsi:106]: Move it to `label` commit.

File: target/linux/ath79/dts/ar9344_wd_mynet-nxxx.dtsi [hunk 26]
```diff
@@ -58,7 +58,7 @@
 				reg = <0x070000 0xf80000>;
 			};
 
-			art: partition@ff0000 {
+			partition@ff0000 {
```
Comment (DragonBluep) [inline target/linux/ath79/dts/ar9344_wd_mynet-nxxx.dtsi:61]: Move it to `label` commit.

File: target/linux/ath79/dts/qca9531_glinet_gl-s200.dtsi [hunk 27]
```diff
@@ -131,7 +131,7 @@
 				reg = <0x040000 0x010000>;
 			};
 
-			art: partition@50000 {
+			partition@50000 {
```
Comment (DragonBluep) [inline target/linux/ath79/dts/qca9531_glinet_gl-s200.dtsi:134]: Move it to `label` commit.

File: target/linux/ath79/dts/qca9558_comfast_cf-e380ac-v2.dts [hunk 28]
```diff
@@ -63,12 +63,6 @@
 	};
 };
 
-&eth0 {
-	status = "okay";
-	mtd-mac-address = <&art 0x0>;
-};
```
Comment (DragonBluep) [inline target/linux/ath79/dts/qca9558_comfast_cf-e380ac-v2.dts:69]: Move it to a new commit.

File: target/linux/ath79/dts/qca955x_senao_router-dual.dtsi [hunk 29]
```diff
@@ -98,7 +98,7 @@
 				read-only;
 			};
 
-			art: partition@ff0000 {
+			partition@ff0000 {
```
Comment (DragonBluep) [inline target/linux/ath79/dts/qca955x_senao_router-dual.dtsi:101]: Move it to `label` commit.

File: target/linux/ath79/dts/qca955x_ubnt_xc.dtsi [hunk 30]
```diff
@@ -53,7 +53,7 @@
 				read-only;
 			};
 
-			art: partition@ff0000 {
+			partition@ff0000 {
```
Comment (DragonBluep) [inline target/linux/ath79/dts/qca955x_ubnt_xc.dtsi:56]: Move it to `label` commit.

File: target/linux/ath79/dts/qca9563_dlink_covr.dtsi [hunk 31]
```diff
@@ -85,7 +85,7 @@
 				reg = <0xe90000 0x160000>;
 			};
 
-			art: partition@ff0000 {
+			partition@ff0000 {
```
Comment (DragonBluep) [inline target/linux/ath79/dts/qca9563_dlink_covr.dtsi:88]: Move it to `label` commit.

File: target/linux/ath79/dts/qca9563_dlink_dir-842-c.dtsi [hunk 32]
```diff
@@ -107,7 +107,7 @@
 				reg = <0x080000 0xf50000>;
 			};
 
-			art: partition@fd0000 {
+			partition@fd0000 {
```
Comment (DragonBluep) [inline target/linux/ath79/dts/qca9563_dlink_dir-842-c.dtsi:110]: Move it to `label` commit.

**Review by DragonBluep — COMMENTED** [review]
Hi! I find some tiny issues.

File: target/linux/ath79/dts/ar1022_sitecom_wlr-7100.dts (hunk 11 above)
Comment (neheb) [reply target/linux/ath79/dts/ar1022_sitecom_wlr-7100.dts:150]: Done.

File: target/linux/ath79/dts/ar9132_tplink_tl-wa901nd-v2.dts (hunk 12 above)
Comment (neheb) [reply target/linux/ath79/dts/ar9132_tplink_tl-wa901nd-v2.dts:90]: Done.

File: target/linux/ath79/dts/qca9563_qxwlan_e1700ac.dtsi (hunk 13 above)
Comment (neheb) [reply target/linux/ath79/dts/qca9563_qxwlan_e1700ac.dtsi:116]: Done.

File: target/linux/ath79/dts/ar7161_buffalo_wzr-hp-ag300h.dtsi (hunk 14 above)
Comment (neheb) [reply target/linux/ath79/dts/ar7161_buffalo_wzr-hp-ag300h.dtsi:149]: Done.

File: target/linux/ath79/dts/ar7240_tplink.dtsi (hunk 16 above)
Comment (neheb) [reply target/linux/ath79/dts/ar7240_tplink.dtsi:102]: Done.

File: target/linux/ath79/dts/ar7240_netgear_wnr612-v2.dtsi (hunk 15 above)
Comment (neheb) [reply target/linux/ath79/dts/ar7240_netgear_wnr612-v2.dtsi:104]: Done.

File: target/linux/ath79/dts/ar724x_ubnt_xm.dtsi (hunk 18 above)
Comment (neheb) [reply target/linux/ath79/dts/ar724x_ubnt_xm.dtsi:61]: Done.

File: target/linux/ath79/dts/qca9563_dlink_dir-842-c.dtsi (hunk 32 above)
Comment (neheb) [reply target/linux/ath79/dts/qca9563_dlink_dir-842-c.dtsi:110]: Done.

File: target/linux/ath79/dts/qca9563_dlink_covr.dtsi (hunk 31 above)
Comment (neheb) [reply target/linux/ath79/dts/qca9563_dlink_covr.dtsi:88]: Done.

File: target/linux/ath79/dts/qca955x_ubnt_xc.dtsi (hunk 30 above)
Comment (neheb) [reply target/linux/ath79/dts/qca955x_ubnt_xc.dtsi:56]: Done.

File: target/linux/ath79/dts/qca955x_senao_router-dual.dtsi (hunk 29 above)
Comment (neheb) [reply target/linux/ath79/dts/qca955x_senao_router-dual.dtsi:101]: Done.

File: target/linux/ath79/dts/qca9531_glinet_gl-s200.dtsi (hunk 27 above)
Comment (neheb) [reply target/linux/ath79/dts/qca9531_glinet_gl-s200.dtsi:134]: Done.

File: target/linux/ath79/dts/ar9344_wd_mynet-nxxx.dtsi (hunk 26 above)
Comment (neheb) [reply target/linux/ath79/dts/ar9344_wd_mynet-nxxx.dtsi:61]: Done.

File: target/linux/ath79/dts/ar9344_tplink_tl-wdrxxxx.dtsi (hunk 25 above)
Comment (neheb) [reply target/linux/ath79/dts/ar9344_tplink_tl-wdrxxxx.dtsi:106]: Done.

File: target/linux/ath79/dts/ar9344_dlink_dir-8x5.dtsi (hunk 23 above)
Comment (neheb) [reply target/linux/ath79/dts/ar9344_dlink_dir-8x5.dtsi:138]: Done.

File: target/linux/ath79/dts/ar9344_qxwlan_e750x.dtsi (hunk 24 above)
Comment (neheb) [reply target/linux/ath79/dts/ar9344_qxwlan_e750x.dtsi:106]: Moved to first commit.

File: target/linux/ath79/dts/ar9344_compex_wpj344-16m.dts (hunk 22 above)
Comment (neheb) [reply target/linux/ath79/dts/ar9344_compex_wpj344-16m.dts:117]: Copy/paste error I think.

File: target/linux/ath79/dts/qca9558_comfast_cf-e380ac-v2.dts (hunk 28 above)
Comment (neheb) [reply target/linux/ath79/dts/qca9558_comfast_cf-e380ac-v2.dts:69]: Done.

File: target/linux/ath79/dts/ar7242_ubnt_sw.dtsi (hunk 17 above)
Comment (neheb) [reply target/linux/ath79/dts/ar7242_ubnt_sw.dtsi:103]: Done.

**Review by DragonBluep — APPROVED** [review]
LGTM.
`Reviewed-by: Shiji Yang <yangshiji66@qq.com>`

**Comment by neheb** [comment]
@DragonBluep added last 3 remaining devices. Now git grep returns only package/kernel/mac80211/patches/ath9k/552-ath9k-ahb_of.patch for mtd-cal-data. I can't tell whether or not that can be removed.

**Comment by DragonBluep** [comment]
> @DragonBluep added last 3 remaining devices. Now git grep returns only package/kernel/mac80211/patches/ath9k/552-ath9k-ahb_of.patch for mtd-cal-data. I can't tell whether or not that can be removed.

This patch contains a lot of other hacks so we can not remove it.

**Comment by neheb** [comment]
Maybe just the mtd-cal-data part?

**Comment by neheb** [comment]
@DragonBluep added extra commit. You mentioned this is nonsensical.

**Comment by DragonBluep** [comment]
> Maybe just the mtd-cal-data part?

Let's wait other people's comments. Personally I'd like to remove it because it is not an upstream feature and we don't need it now.

> @DragonBluep added extra commit. You mentioned this is nonsensical.

Still LGTM.

**Comment by neheb** [comment]
ping @hauke
